cluster.py

```diff
@@ -8,14 +10,33 @@
 logger = logging.getLogger(__name__)
 
 
+def get_cluster_lock(timeout: int = 10 * 60) -> socket.socket:
+    logger.info("Acquiring cluster lock to prevent concurrent run on the same machine...")
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    start_time = time.time()
+    while start_time + timeout > time.time():
+        try:
+            sock.bind(('127.0.0.1', 48783))  # random port
+            logger.info("Cluster lock acquired")
+            return sock
+        except OSError:
+            time.sleep(1)
+    raise TimeoutError(f"Couldn't acquire cluster lock within {timeout} seconds")
+
+
+def release_cluster_lock(sock: socket.socket) -> None:
+    sock.close()
+
+
 class TestCluster:
     """Responsible for configuring, starting and stopping cluster for tests"""
 
     def __init__(self, driver_directory: Path, version: str, configuration: Dict[str, str]) -> None:
         self.cluster_directory = driver_directory / "ccm"
         self.cluster_directory.mkdir(parents=True, exist_ok=True)
         logger.info("Preparing test cluster binaries and configuration...")
-        self._cluster: ccm.ScyllaCluster = ccm.ScyllaCluster(self.cluster_directory, 'test', cassandra_version=version)
+        self._cluster_lock = get_cluster_lock()
+        self._cluster: ccm.ScyllaCluster = ccm.ScyllaCluster(self.cluster_directory, f'test', cassandra_version=version)
```

Comment:
Looks like redundant change.
Also, it must violate pylint rules - the `f` string doesn't use substitution.

Reply:
yes, redundant, leftover from other approach.
This repo does not support pylint yet.
Fixed.